lib/cocoapods/installer/target_installer/pod_target_installer.rb

```diff
@@ -66,9 +67,13 @@ def add_files_to_build_phases
           native_target.add_file_references(header_file_refs) do |build_file|
             # Set added headers as public if needed
             if target.requires_frameworks?
+              build_file.settings ||= {}
               if public_headers.include?(build_file.file_ref.real_path)
-                build_file.settings ||= {}
                 build_file.settings['ATTRIBUTES'] = ['Public']
+              elsif private_headers.include?(build_file.file_ref.real_path)
+                build_file.settings['ATTRIBUTES'] = ['Private']
+              else
+                build_file.settings['ATTRIBUTES'] = ['Project']
```

Comment:
I'm unsure about this, as we only ever define either public or private headers in a specification. If this is really needed, we should have a `project_headers` attribute in the DSL to explicitly define them as well.